### PR TITLE
Feature/user registration

### PR DESCRIPTION
**Pull request by marvindore**
Create controller, entities, and tests for the Passenger service.

### CODE REVIEW

**Review by jusoto — APPROVED** [review]
Looking good

**Comment by Castyr** [comment]
[Implement the User Registration Use Case](https://trello.com/c/CmSzCKaw/100-implement-the-user-registration-use-case)

File: README.md [hunk 1]
```diff
@@ -1,2 +1,124 @@
-# passenger
+# Passenger CRUD Service
```
Comment (Castyr) [inline README.md]: Let's just call this one Passenger Service as it may have more features down the road.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,124 @@
...
+
+The following applications must have been installed in your system previous to run this application.
+
+ - Java8
+ - Postgresql
+ 
+#### Installing
+
+Check Postgresql root and username and password on file:
+   src/main/resources/application.properties
```
Comment (Castyr) [inline README.md]: Keep this standardized with the rest of the services, convert it to YAML.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,124 @@
...
+## Deployment
+
+-
+
+## Built With
+
+* [Spring Boot](https://spring.io/docs) - Application Framework
+* [Gradle](https://docs.gradle.org/4.2/release-notes.html) - Dependency Management
+
+## Authors
```
Comment (Castyr) [inline README.md]: This is somewhat redundant as the commits will have your user on them.

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+#  user-service:
+#    image: aista/user-service
+#    container_name: user-service
+#    ports:
+#      - '8091:8091'
+#    depends_on:
+#      - mysql-server
+#      - discovery-service
+#
+#  mysql-server:
```
Comment (Castyr) [inline docker-compose.yml]: Why is there also a MySQL container here?

File: src/main/java/org/aitesting/passenger/controllers/PassengerController.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+public class PassengerController {
+
+
+    @Autowired
+    private PassengerService passengerService;
+
+    private ModelMapper modelMapper = new ModelMapper();
+
+
+    @GetMapping(value="passenger")
```
Comment (Castyr) [inline src/main/java/org/aitesting/passenger/controllers/PassengerController.java]: In the GetMapping annotations "passenger" is redundant, you could move it up to the class.

File: src/main/java/org/aitesting/passenger/controllers/PassengerController.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    public List<Passenger> getAllPassengers() {
+        return passengerService.getList();
+    }
+
+    @GetMapping(value="passenger/{passID}")
+    public Passenger getPassenger(@PathVariable long passID) {
+        return passengerService.getPassenger(passID);
```
Comment (Castyr) [inline src/main/java/org/aitesting/passenger/controllers/PassengerController.java]: You should return EntityResponse<Passenger> or similar objects with a Http status code. What if there is no passenger by the given ID? In this case a 404 should be given.

File: src/main/java/org/aitesting/passenger/controllers/PassengerController.java [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+    @Autowired
+    private PassengerService passengerService;
+
+    private ModelMapper modelMapper = new ModelMapper();
+
+
+    @GetMapping(value="passenger")
+    public List<Passenger> getAllPassengers() {
+        return passengerService.getList();
```
Comment (Castyr) [inline src/main/java/org/aitesting/passenger/controllers/PassengerController.java]: Here EntityResponse<List<Passenger>> should be the return type.

File: src/main/java/org/aitesting/passenger/controllers/PassengerController.java [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+        }
+        catch(Exception e){
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                    .body("Passenger cannot be created");
+        }
+    }
+
+    @PutMapping(value="passenger")
+    public void updatePassenger(@RequestBody Passenger passenger){
+        passengerService.updatePassenger(passenger);
```
Comment (Castyr) [inline src/main/java/org/aitesting/passenger/controllers/PassengerController.java]: Update and Delete should have 404 paths too.

File: src/main/java/org/aitesting/passenger/interfaces/Command.java [hunk 9]
```diff
@@ -0,0 +1,5 @@
+package org.aitesting.passenger.interfaces;
+
+public interface Command {
```
Comment (Castyr) [inline src/main/java/org/aitesting/passenger/interfaces/Command.java]: This interface is not needed

File: src/main/java/org/aitesting/passenger/services/PassengerService.java [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+    public Passenger getPassenger(long id){
+        return passengerRepository.findOne(id);
+    }
+
+    public void addPassenger(Passenger passenger){
+        passengerRepository.save(passenger);
+    }
+
+    public void updatePassenger(Passenger passenger){
+        passengerRepository.save(passenger);
```
Comment (Castyr) [inline src/main/java/org/aitesting/passenger/services/PassengerService.java]: This method has a side effect of creating a passenger when it doesn't exist, you need to check that it exists first.

File: src/test/java/org/aitesting/passenger/provider/PassengerContractBase.java [hunk 11]
```diff
@@ -0,0 +1,52 @@
+package org.aitesting.passenger.provider;
+
+
+//import com.jayway.restassured.module.mockmvc.RestAssuredMockMvc;
```
Comment (Castyr) [inline src/test/java/org/aitesting/passenger/provider/PassengerContractBase.java]: Don't check in commented code.

File: src/test/java/org/aitesting/passenger/provider/PassengerContractBase.java [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+@RunWith(SpringRunner.class)
+@ActiveProfiles("test")
+@SpringBootTest(classes = PassengerApplication.class)
+public abstract class PassengerContractBase {
+
+    @Autowired
+    private WebApplicationContext webApplicationContext;
+
+    @Autowired
+    PassengerController passengerController;
```
Comment (Castyr) [inline src/test/java/org/aitesting/passenger/provider/PassengerContractBase.java]: Don't leave out encapsulation keywords.

File: src/main/java/org/aitesting/passenger/model/PassengerDto.java [hunk 13]
```diff
@@ -0,0 +1,35 @@
+package org.aitesting.passenger.model;
```
Comment (Castyr) [inline src/main/java/org/aitesting/passenger/model/PassengerDto.java]: This DTO is not needed, this is exactly what the Passenger model is.

File: src/main/java/org/aitesting/passenger/model/Passenger.java [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+                     String city, String state, String zip, String phoneNumber){
+        this.firstName = firstName;
+        this.lastName = lastName;
+        this.email = email;
+        this.streetAddress = streetAddress;
+        this.city = city;
+        this.state = state;
+        this.zip = zip;
+        this.phoneNumber = phoneNumber;
+    }
```
Comment (Castyr) [inline src/main/java/org/aitesting/passenger/model/Passenger.java:59]: Where are your getters?

**Review by Castyr — COMMENTED** [review]
Fix the commented stuff.

File: src/main/java/org/aitesting/passenger/model/PassengerDto.java (hunk 13 above)
Comment (marvindore) [reply src/main/java/org/aitesting/passenger/model/PassengerDto.java]: Wouldn't the DTO be needed for POST request and the Passenger model for saving to the database?

File: src/main/java/org/aitesting/passenger/model/Passenger.java (hunk 14 above)
Comment (marvindore) [reply src/main/java/org/aitesting/passenger/model/Passenger.java:59]: I am using the project lombok library, it creates the getters and setters at compile time.

**Review by Castyr — APPROVED** [review]

**Pull request merged**